src/dex/woo-fi/woo-fi.ts

```diff
@@ -0,0 +1,556 @@
+import _ from 'lodash';
+import { Interface, Result } from '@ethersproject/abi';
+import {
+  Token,
+  Address,
+  ExchangePrices,
+  AdapterExchangeParam,
+  SimpleExchangeParam,
+  PoolLiquidity,
+  Logger,
+} from '../../types';
+import { SwapSide, Network, NULL_ADDRESS } from '../../constants';
+import { getBigIntPow, getDexKeysWithNetwork, wrapETH } from '../../utils';
+import { IDex } from '../../dex/idex';
+import { IDexHelper } from '../../dex-helper/idex-helper';
+import { PoolState, WooFiData } from './types';
+import { SimpleExchange } from '../simple-exchange';
+import { WooFiConfig, Adapters } from './config';
+import { wooFiMath } from './woo-fi-math';
+import {
+  MIN_CONVERSION_RATE,
+  USD_PRECISION,
+  WOO_FI_GAS_COST,
+} from './constants';
+import wooPPABI from '../../abi/woo-fi/WooPP.abi.json';
+import wooFeeManagerABI from '../../abi/woo-fi/WooFeeManager.abi.json';
+import woOracleABI from '../../abi/woo-fi/Wooracle.abi.json';
+
+export class WooFi extends SimpleExchange implements IDex<WooFiData> {
+  readonly math: typeof wooFiMath = wooFiMath;
+
+  latestState: PoolState | null = null;
+
+  latestBlockNumber: number = 0;
+
+  tokenByAddress: Record<string, Token> | null = null;
+
+  readonly wooIfaces = {
+    PP: new Interface(wooPPABI),
+    fee: new Interface(wooFeeManagerABI),
+    oracle: new Interface(woOracleABI),
+  };
```

Comment:
you can make these static to be used across networks?

Reply:
Fixed